riscv/csrs.cc

```diff
@@ -1400,3 +1412,39 @@ void henvcfg_csr_t::verify_permissions(insn_t insn, bool write) const {
       throw trap_illegal_instruction(insn.bits());
   }
 }
+
+bool henvcfg_csr_t::unlogged_write(const reg_t val) noexcept {
+  // mask out read-only 0 bits if menvcfg has them turned off
+  return masked_csr_t::unlogged_write((menvcfg->read() | ~(MENVCFG_PBMTE | MENVCFG_STCE)) & val);
+}
+
+stimecmp_csr_t::stimecmp_csr_t(processor_t* const proc, const reg_t addr, const reg_t imask):
+  basic_csr_t(proc, addr, 0), intr_mask(imask) {
+}
+
+bool stimecmp_csr_t::unlogged_write(const reg_t val) noexcept {
+  state->mip->backdoor_write_with_mask(intr_mask, 0);
+  // time will return time + htimedelta if v = true
+  if (state->time->read() >= val) {
+    state->mip->backdoor_write_with_mask(intr_mask, intr_mask);
+  }
```

Comment:
Instead of sometimes doing two writes to `mip`, compute one value (either `intr_mask` or 0) and do one write every time.
```
bool pending = state->time->read() >= val;
state->mip->backdoor_write_with_mask(intr_mask, pending ? intr_mask : 0);
```